### PR TITLE
different # input channels should retain pre-trained weights 

### LINKED ISSUE #508: different # input channels should retain pre-trained weights
we can average pre-trained weights and copy them across the [1] dimension of resnet's .conv1 in the case of a resnet. Other arch. may require different syntax. 

resnet implementation in branch "feat_conv1_weights" but this branch may not get merged to develop (it branches off "feat_wandb")

### PR DESCRIPTION
**Pull request by sammlapp**
resolves #508 (for ResNet architectures only):
#508 states that if the number of input channels to a CNN is different from a pretrained network (eg ImageNet with 3 channels), it should still be able to use pre-trained weights for conv1. 
It accomplishes this for ResNet architectures by copying the mean value of conv1 layer weights across channels to all channels, if the number of channels is not 3. If the number of channels is 3, the original weights are used rather than applying the average value to 3 channels.

### CODE REVIEW

**Comment by sammlapp** [comment]
@louisfh in reviewing this, let me know if you think this is a sensible solution or if you see any problems with it. The averaging weights over channels is something others have done, and seemed to work pretty well for me. You don't need to test the performance.

**Comment by sammlapp** [comment]
FYI, I added some documentation updates (closing various issues) here rather than creating separate branches for each one.

**Comment by louisfh** [comment]
I'm getting a strange error in the `preprocessors.ipynb` notebook. This is in a fresh py3.8 environment using poetry.  Here's the stack:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/var/folders/_s/zm3gz4x52glfbflc25_3rp3w0000gq/T/ipykernel_56092/3868426432.py in <module>
----> 1 pre = SpectrogramPreprocessor(sample_duration=2.0)

~/Dev/opensoundscape/opensoundscape/preprocess/preprocessors.py in __init__(self, sample_duration, overlay_df, out_shape)
    210         self.pipeline = pd.Series(
    211             {
--> 212                 "load_audio": AudioClipLoader(),
    213                 # if we are augmenting and get a long file, take a random trim from it
    214                 "random_trim_audio": AudioTrim(is_augmentation=True, random_trim=True),

~/Dev/opensoundscape/opensoundscape/preprocess/actions.py in __init__(self, **kwargs)
    137 
    138     def __init__(self, **kwargs):
--> 139         super(AudioClipLoader, self).__init__(
    140             Audio.from_file, extra_args=["_start_time", "_sample_duration"], **kwargs
    141         )

~/Dev/opensoundscape/opensoundscape/preprocess/actions.py in __init__(self, fn, is_augmentation, extra_args, **kwargs)
     98 
     99         # update self.params with any user-provided parameters
--> 100         self.set(**kwargs)
    101 
    102         # make sure all required args are given (skipping the first, which will be provided by go)

~/Dev/opensoundscape/opensoundscape/preprocess/actions.py in set(self, **kwargs)
     56             []
     57         ), f"unexpected arguments: {unmatched_args}. The valid arguments and current values are: \n{self.params}"
---> 58         self.params.update(kwargs)
     59 
     60     def get(self, arg):

/opt/anaconda3/envs/opso_py38_poetry/lib/python3.8/site-packages/pandas/core/series.py in update(self, other)
   3245 
   3246         if not isinstance(other, Series):
-> 3247             other = Series(other)
   3248 
   3249         other = other.reindex_like(self)

/opt/anaconda3/envs/opso_py38_poetry/lib/python3.8/site-packages/pandas/core/series.py in __init__(self, data, index, dtype, name, copy, fastpath)
    355             if is_empty_data(data) and dtype is None:
    356                 # gh-17261
--> 357                 warnings.warn(
    358                     "The default dtype for empty Series will be 'object' instead "
    359                     "of 'float64' in a future version. Specify a dtype explicitly "

TypeError: issubclass() arg 2 must be a class or tuple of classes
```

**Comment by louisfh** [comment]
If you comment out `warnings.simplefilter('ignore',category='FutureWarning')` the notebook executes fine.

File: opensoundscape/preprocess/actions.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class BaseAction:
     """
 
     def __init__(self):
-        self.params = pd.Series(dtype=object)
```
Comment (louisfh) [inline opensoundscape/preprocess/actions.py:36]: We should make this change throughout the codebase to be consistent if we need to.

**Comment by louisfh** [comment]
We should add to the documentation that weights are averaged across channels when number of input channels != 3.

**Pull request merged**